Dynlink not built if both `--disable-debugger` and `--disable-ocamldoc` are used

What it says on the tin. If both flags are passed, then `make install` fails due to a missing `dynlink.cma`:

```
$ make install
[...]
rm -f "/home/nojebar/ocaml/local/lib/ocaml"/dynlink.cm* "/home/nojebar/ocaml/local/lib/ocaml/dynlink.mli" \
        "/home/nojebar/ocaml/local/lib/ocaml/dynlink.a" \
        "/home/nojebar/ocaml/local/lib/ocaml/"dynlink_config.cmx "/home/nojebar/ocaml/local/lib/ocaml/"dynlink_types.cmx "/home/nojebar/ocaml/local/lib/ocaml/"dynlink_platform_intf.cmx "/home/nojebar/ocaml/local/lib/ocaml/"dynlink_common.cmx "/home/nojebar/ocaml/local/lib/ocaml/"dynlink.cmx
mkdir -p "/home/nojebar/ocaml/local/lib/ocaml/dynlink"
/usr/bin/install -c -p -m 644 \
  otherlibs/dynlink/dynlink.cmi otherlibs/dynlink/dynlink.cma \
  otherlibs/dynlink/META \
  "/home/nojebar/ocaml/local/lib/ocaml/dynlink"
/usr/bin/install: cannot stat 'otherlibs/dynlink/dynlink.cma': No such file or directory
make: *** [Makefile:2688: install] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -2070,13 +2070,11 @@ ocamltest/ocamltest.html: ocamltest/ocamltest.org
 # The extra libraries
 
 .PHONY: otherlibraries
-otherlibraries: ocamltools
-	$(MAKE) dynlink-all
+otherlibraries: ocamltools dynlink-all
 	$(MAKE) -C otherlibs all
 
 .PHONY: otherlibrariesopt
-otherlibrariesopt:
-	$(MAKE) dynlink-allopt
+otherlibrariesopt: dynlink-allopt
 	$(MAKE) -C otherlibs allopt
 
 otherlibs/unix/unix.cmxa: otherlibrariesopt

--- a/Makefile
+++ b/Makefile
@@ -2071,10 +2071,12 @@ ocamltest/ocamltest.html: ocamltest/ocamltest.org
 
 .PHONY: otherlibraries
 otherlibraries: ocamltools
+	$(MAKE) dynlink-all
 	$(MAKE) -C otherlibs all
 
 .PHONY: otherlibrariesopt
 otherlibrariesopt:
+	$(MAKE) dynlink-allopt
 	$(MAKE) -C otherlibs allopt
 
 otherlibs/unix/unix.cmxa: otherlibrariesopt

